metapool/sample_sheet.py

```diff
@@ -1237,38 +1237,93 @@ def __init__(self, path=None):
         }
 
 
-def load_sample_sheet(sample_sheet_path):
-    # Load the sample-sheet using various KLSampleSheet children and return
-    # the first instance that produces a valid sample-sheet. We assume that
-    # because of specific SheetType and SheetVersion values, no one sample-
-    # sheet can match more than one KLSampleSheet child.
+def _parse_header(fp):
+    df = pd.read_csv(fp, dtype="str", sep=",", header=None,
+                     names=range(100))
```

Comment:
My concern with this approach is that IMOO this could make a lot of erroneous sample_sheets pass this method but they will fail in the other validations downstream; do you think this could be a problem?

Reply:
No, that won't be a problem. I could add a check to make sure the three attributes I am querying are present, instead of assuming that they are. Other than that, we want it to pass _parse_header() successfully. The reason why is now that we have different sample-sheet versions, we need a point-of-reference when validating them. For instance, if the [Data] section doesn't contain the 'Sample_Well' column, that's expected for legacy metagenomic sheets but not for current sample-sheets. Without knowing what kind of sheet this is, it's no longer possible to validate it with certainty.

The con of this method is obviously we create a large initial dataframe only to immediately pare it down. The pros of this method is legacy sample-sheets with header comments like these:
```
# PI,Knight,robknight@ucsd.edu,,,,,,,,
# Contact,Gail Ackermann,Greg Humphrey,Jeff Dereus,Jon Sanders,,,,,,
# ,ackermag@ucsd.edu,ghsmu414@gmail.com,jdereus@ucsd.edu,jonsan@gmail.com,,,,,,
```
don't cause `ParserError: Error tokenizing data. C error: Expected 11 fields in line 4, saw 12`.

Also, read_csv() appears to use the number of columns implied in row 1 to determine the shape of the dataframe. I wanted to see what the effect would be if I had an otherwise correct sample-sheet, but I made the first column `[Header],,,,,`. The results were that since the rest of the sheet has 12 columns, the first six columns were silently mashed to create the index for a row, and the final six became the values for the six columns. This means that legacy sample-sheets missing just a single ',' will silently load in an unpredictable way. By making the dataframe arbitrarily large, we would be able to avoid this.